src/Microsoft.AspNetCore.Mvc.Core/Internal/ObjectMethodExecutor.cs

```diff
@@ -238,5 +254,43 @@ private static Task<object> Convert<T>(object taskAsObject)
             var task = (Task<T>)taskAsObject;
             return CastToObject<T>(task);
         }
+
+        private void EnsureParameterDefaultValues()
+        {
+            if (_parameterDefaultValues == null)
+            {
+                var count = ActionParameters.Length;
+                _parameterDefaultValues = new object[count];
+
+                for (var i = 0; i < count; i++)
+                {
+                    var parameterInfo = ActionParameters[i];
+                    object defaultValue;
+
+                    if (parameterInfo.HasDefaultValue)
+                    {
+                        defaultValue = parameterInfo.DefaultValue;
+                    }
+                    else
+                    {
+                        var defaultValueAttribute = parameterInfo
+                            .GetCustomAttribute<DefaultValueAttribute>(inherit: false);
+
+                        if (defaultValueAttribute?.Value == null)
+                        {
+                            defaultValue = parameterInfo.ParameterType.GetTypeInfo().IsValueType
```

Comment:
What's the concern? perf or correctness?

I'm not concerned about the perf of this, it's done once, that's the point of this change :laughing: